stonybook/pipeline/quote/quote.py

```diff
@@ -0,0 +1,961 @@
+            for j, quoteObj in enumerate(conversation):
+                if quoteObj.mention == None:
+                    status = self.searchUpwards(i, j)
+                    if status == False:
+                        self.searchDownwards(i, j)
+        self.updateQuotes()
+
+    def updateQuotes(self):
+        for conversation in self.conversations:
+            for quoteObj in conversation:
+                self.quotes[quoteObj.quoteListIdx].mention = quoteObj.mention
+                self.quotes[quoteObj.quoteListIdx].mention_start = quoteObj.mention_start
+                self.quotes[quoteObj.quoteListIdx].mention_end = quoteObj.mention_end
+
+    def populateQuoteListIdx(self):
+        for i, quoteObj in enumerate(self.quotes):
+            self.quotes[i].quoteListIdx = i
+
+    def looseConversation(self):
+        for quoteObj in self.quotes:
+            if quoteObj.mention is None:
+                mention = None
+                # Search Upwards
+                idx = quoteObj.quoteListIdx - 1
+                while (idx >= 0):
+                    if self.quotes[idx].mention is not None:
+                        self.quotes[quoteObj.quoteListIdx].mention = self.quotes[idx].mention
+                        self.quotes[quoteObj.quoteListIdx].mention_start = self.quotes[idx].mention_start
+                        self.quotes[quoteObj.quoteListIdx].mention_end = self.quotes[idx].mention_end
+                        break
+                    idx = idx - 1
+
+    def dumpXML(self, input_xml_path, output_xml_path):
+        # Writes deterministic sieves output to character coref xml file
+
+        tree = ET.parse(input_xml_path)
+        root = tree.getroot()
+
+        # Temporary quote (for syntax)
+        temp_quote = root.find('./analysis/quotes/quote')
+
+        # To remove the existing quotes information
+        for analysis_1 in root.iter('analysis'):
+            for quotes_1 in analysis_1.iter('quotes'):
+                analysis_1.remove(quotes_1)
+
+        # Added quotes empty tag
+        for analysis in root.iter('analysis'):
+            ET.SubElement(analysis, "quotes").text = "\n"
+
+        # Added quote tags
+        for analysis_2 in root.iter('analysis'):
+            for quotes_2 in analysis_2.iter('quotes'):
+                for quote_2 in self.quotes:
+                    ET.SubElement(quotes_2, "quote",
+                                  mention=quote_2.mention,
+                                  tokBegin=str(quote_2.quote_start),
+                                  tokEnd=str(quote_2.quote_end),
+                                  mentionBegin=str(quote_2.mention_start),
+                                  mentionEnd=str(quote_2.mention_end))
+
+        tree.write(output_xml_path, encoding='utf-8' ,pretty_print=True)
+
+def run(book_dir, input_xml_path, output_xml_path):
+    tokens_list = getTokenList(input_xml_path)
+    tokens_list = normalizeQuotes(tokens_list)
```

Comment:
In the pipeline, we already call unidecode before passing through. Can you verify if this is actually modifying the token list when you call it?